web/packages/shared/components/AccessRequests/NewRequest/RequestCheckout/RequestCheckout.tsx

```diff
@@ -208,131 +233,178 @@ export function RequestCheckout<T extends PendingListItem>({
+  function customRow(item: T) {
+    if (item.kind === 'kube_cluster') {
+      return (
+        <td colSpan={3}>
+          <Flex>
+            <Flex flexWrap="wrap">
+              <Flex
+                gap={2}
+                justifyContent="space-between"
+                width="100%"
+                alignItems="center"
+              >
+                <Flex gap={5}>
+                  <Box>{getPrettyResourceKind(item.kind)}</Box>
+                  <Box>{item.name}</Box>
                 </Flex>
-              )}
-              {userGroupFetchAttempt?.status === 'failed' && (
-                <Danger mt={4}>{userGroupFetchAttempt.statusText}</Danger>
-              )}
-              {userGroupFetchAttempt?.status === 'success' &&
-                appsGrantedByUserGroup.length > 0 && (
-                  <AppsGrantedAccess apps={appsGrantedByUserGroup} />
-                )}
-              {isResourceRequest && (
-                <ResourceRequestRoles
-                  roles={resourceRequestRoles}
-                  selectedRoles={selectedResourceRequestRoles}
-                  setSelectedRoles={setSelectedResourceRequestRoles}
-                  fetchAttempt={fetchResourceRequestRolesAttempt}
+                <CrossIcon
+                  clearAttempt={clearAttempt}
+                  item={item}
+                  toggleResource={toggleResource}
+                  createAttempt={createAttempt}
                 />
-              )}
-              <Box mt={6} mb={1}>
-                <SelectReviewers
-                  reviewers={dryRunResponse?.reviewers.map(r => r.name) ?? []}
-                  selectedReviewers={selectedReviewers}
-                  setSelectedReviewers={setSelectedReviewers}
-                />
-              </Box>
-              <Validation>
-                {({ validator }) => (
+              </Flex>
+              <KubeNamespaceSelector
+                kubeClusterItem={item}
+                savedResourceItems={data}
+                toggleResource={toggleResource}
+                fetchKubeNamespaces={fetchKubeNamespaces}
+                bulkToggleKubeResources={bulkToggleKubeResources}
+                namespaceRequired={
+                  requiresNamespaceSelect &&
+                  affectedKubeClusterName.includes(item.id)
+                }
+              />
+            </Flex>
+          </Flex>
+        </td>
+      );
+    }
+  }
+
+  return (
+    <Validation>
+      {({ validator }) => (
+        <>
+          {showRequestRoleErrBanner &&
+            fetchResourceRequestRolesAttempt.status === 'failed' && (
+              <Alert
+                kind="danger"
+                children={fetchResourceRequestRolesAttempt.statusText}
+              />
+            )}
+          {hasUnsupportedKubeRequestModes && (
+            <Alert kind="danger">
+              You can only request Kubernetes resource kind{' '}
+              {unsupportedKubeRequestModes} for cluster{' '}
```

Comment:
Is there any way to get rid of those square brackets around resource kinds? How is the message going to look like if `unsupportedKubeRequestModes` includes multiple kinds?

Reply:
it'll look like this `[secret pod namespace]`, opting not to change anything, since it follows error messaging in tsh

Reply:
That's fine for now, but long term I think we should strive to not leak too many CLI-specific conventions leak to GUIs. I had a similar discussion with Grzegorz recently. https://github.com/gravitational/teleport/pull/47652#discussion_r1809112242

I'm not sure why tsh shows the resource kinds in square brackets in the first place, but in the UI it looks kind of weird IMHO – we have much more freedom there with regards to styling when it comes showing data like this. TBH I think just a regular string would be fine here, e.g. "You can only request Kubernetes resource kinds pod and secret", where "resource kinds" gets pluralized depending on the count and the array with allowed kinds uses something like [`Array#to_sentence` from Ruby on Rails](https://api.rubyonrails.org/classes/Array.html#method-i-to_sentence).

Reply:
Oh, I keep forgetting that there's now an equivalent of `Array#to_sentence` in JS. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat

```javascript
const vehicles = ['Motorcycle', 'Bus', 'Car'];

const formatter = new Intl.ListFormat('en', {
  style: 'long',
  type: 'conjunction',
});
console.log(formatter.format(vehicles));
// Expected output: "Motorcycle, Bus, and Car"
```

Though I'm not sure if this is something we could use without a polyfill. caniuse.com says that it's not available at all in Safari on Macs that are not running Big Sur or better. https://caniuse.com/mdn-javascript_builtins_intl_listformat

Reply:
i hear you, i created a function for it here: https://github.com/gravitational/teleport/pull/47345/commits/a7512049b2755800ae4e90e7ef8def7ea653715a

result:

<img width="492" alt="image" src="https://github.com/user-attachments/assets/82c9aaca-394d-4912-a606-e053b3f33a13">

<img width="492" alt="image" src="https://github.com/user-attachments/assets/0d84f00b-cc97-41b2-b931-e24a2a16c2fa">

<img width="492" alt="image" src="https://github.com/user-attachments/assets/8c96272a-5024-49fa-a966-6ef132ae6788">